app/jobs/full_text_to_solr_job.rb

```diff
@@ -2,24 +2,33 @@
 
 ##
 # a job class that will update a Solr document
-# with the extracted text of an attached PDF
+# with the extracted text of  attached PDFs
 class FullTextToSolrJob < ApplicationJob
   # @return [String] the id for the SolrDocument being updated
   # @return [IO] the PDF content
-  attr_reader :solr_document_id, :work_content
+  attr_reader :work_id
 
   queue_as :default
 
-  def initialize(solr_document_id, work_content)
-    @solr_document_id = solr_document_id
-    @work_content = work_content
+  def initialize(work_id)
+    @work_id = work_id
   end
 
   def perform
     solr = RSolr.connect(url: ActiveFedora::SolrService.instance.conn.uri.to_s)
-    solr_document = SolrDocument.find(@solr_document_id).to_h
-    solr_document["all_text_timv"] = FullTextExtractor.new(@work_content).text
+    solr_document = SolrDocument.find(@work_id).to_h
+    solr_document["all_text_timv"] = all_text
     solr.add(solr_document)
     solr.commit
   end
+
+  private
+
+    def files
+      ActiveFedora::Base.find(@work_id).file_sets[0].files
```

Comment:
Each supplemental file is its own FileSet: https://github.com/ucsblibrary/alexandria/blob/master/lib/importer/factory/etd_factory.rb#L37-L52

So this will just pick one (possibly at random, given how RDF doesn't enforce order) to index.

I'm not sure how the searching will look if we smush everything together; would it work to simply do `ActiveFedora::Base.find(@work_id).file_sets.map(&:files).flatten`?

Reply:
Sorry! `FileSet`s throw me off. 

Smushing everything together is what we want, in the search results it just returns the work.